packages/query-core/src/__tests__/query.test.tsx

```diff
@@ -943,4 +943,39 @@ describe('query', () => {
     await sleep(60) // let it resolve
     expect(spy).toHaveBeenCalledWith('1 - 2')
   })
+
+  it('should call onError when setData throws', async () => {
+    const key = queryKey()
+
+    const queryFn = vi.fn<Array<unknown>, unknown>()
+
+    queryFn.mockImplementation(async () => {
+      await sleep(50)
```

Comment:
let's get those numbers down:

```suggestion
      await sleep(10)
```